File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file111.cpp
#include <iostream>
#include <omp.h>

int main() {
    int var[8];
    int i, j;

    // Initialize var array
    for (i = 0; i < 8; i++) {
        var[i] = 0;
    }

    // OpenMP target region
    #pragma omp target teams distribute parallel do reduction(+:var[0:7])
    for (i = 0; i < 20; i++) {
        #pragma omp simd
        for (j = 0; j < 8; j++) {
            var[j]++;
        }
    }

    // OpenMP end target region

    // Check if var array is correct
    for (i = 0; i < 8; i++) {
        if (var[i]!= 20) {
            std::cout << var[i] << std::endl;
        }
    }

    return 0;
}

File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file111.f90
program DRB162_nolocksimd_orig_gpu_no
    use omp_lib
    implicit none

    integer :: var(8)
    integer :: i,j

    do i = 1, 8
        var(i) = 0
    end do

    !$omp target map(tofrom:var) device(0)
    !$omp teams num_teams(1) thread_limit(1048)
    !$omp distribute parallel do reduction(+:var)
    do i = 1, 20
        !$omp simd
        do j = 1, 8
            var(j) = var(j)+1
        end do
        !$omp end simd
    end do
    !$omp end distribute parallel do
    !$omp end teams
    !$omp end target

    do i = 1, 8
        if (var(i) /= 20) then
            print*,var(i)
        end if
    end do

end program
